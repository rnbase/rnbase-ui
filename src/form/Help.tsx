import React from 'react';
import { Platform, StyleSheet, Text, TextProps } from 'react-native';

import { Theme, Themed, withTheme } from '../theming';

interface HelpProps extends TextProps {
  children: React.ReactNode;
}

const Help: React.FC<Themed<typeof createStyleSheet, HelpProps>> = ({
  theme: { styles },
  children,
  style,
  ...props
}) => (
  <Text {...props} style={[styles.root, style]}>
    {children}
  </Text>
);

const createStyleSheet = ({ colors, fonts }: Theme) =>
  StyleSheet.create({
    root: {
      ...fonts.normal,
      margin: 15,
      marginTop: 0,
      fontSize: 12,
      color: colors.gray2,
      lineHeight: Platform.select({
        ios: 14,
        android: 18,
      }),
    },
  });

export default withTheme(Help, createStyleSheet, 'Form.Help');
