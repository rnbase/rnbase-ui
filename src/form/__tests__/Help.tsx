import React from 'react';

// Note: test renderer must be required after react-native.
import TestRenderer from 'react-test-renderer';

import Help from '../Help';

type Props = React.ComponentProps<typeof Help>;

/**
 * Under test
 */
const createElement = (props: Partial<Props>) => <Help {...props}>help</Help>;

const createRenderer = (props: Partial<Props>) => TestRenderer.create(createElement(props));

it('should render normally', () => {
  expect(createRenderer({})).toMatchSnapshot();
});

it('should render with custom styles', () => {
  expect(createRenderer({ style: { color: 'red' } })).toMatchSnapshot();
});
