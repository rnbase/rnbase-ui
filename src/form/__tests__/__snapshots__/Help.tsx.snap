// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render normally 1`] = `
<Text
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "color": "#aeaeb2",
        "fontFamily": "System",
        "fontSize": 12,
        "fontWeight": "400",
        "includeFontPadding": false,
        "lineHeight": 14,
        "margin": 15,
        "marginTop": 0,
        "textAlignVertical": "center",
      },
      undefined,
    ]
  }
>
  help
</Text>
`;

exports[`should render with custom styles 1`] = `
<Text
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "color": "#aeaeb2",
        "fontFamily": "System",
        "fontSize": 12,
        "fontWeight": "400",
        "includeFontPadding": false,
        "lineHeight": 14,
        "margin": 15,
        "marginTop": 0,
        "textAlignVertical": "center",
      },
      Object {
        "color": "red",
      },
    ]
  }
>
  help
</Text>
`;
